jsonschema/src/keywords/custom_keyword_schema.rs

```diff
@@ -0,0 +1,101 @@
+use crate::compilation::context::CompilationContext;
+use crate::compilation::options::KeywordDefinition;
+use crate::error::no_error;
+use crate::keywords::CompilationResult;
+use crate::paths::{InstancePath, JSONPointer, PathChunk};
+use crate::validator::Validate;
+use crate::{ErrorIterator, JSONSchema, ValidationError};
+use serde_json::{Map, Value};
+use std::fmt::{Display, Formatter};
+
+pub(crate) struct CustomKeywordSchemaValidator {
+    schema_path: JSONPointer,
+    keyword_schema: Option<Value>,
```

Comment:
What would be the use case for `None`? 

As far as I can tell, if one wants to create a custom keyword, then there is always some value there, even if it is a marker-like `null` value. I don't think that there is a place for niche optimization for `Option<Value>`, so there should be no memory win there anyway. 

Therefore I am thinking that the plain `Value` could be the way to go here. What do you think?